Update VS Command line HowTo

BillWagner opened the pull request:
The VSVars.bat is now VsDevCmd.bat.

Fixes #2287

BillWagner committed:
Updatee VS Command line HowTo

The VSVars.bat is not VsDevCmd.bat.

mairaw commented:
Given that these topics can also be used by VS 2015 customers, should you add a note that the file was renamed in 2017?

BillWagner committed:
add text for previous versions

Earlier versions of Visual Studio used a different file name. Make sure we still support that.

BillWagner commented:
Good point @mairaw I've updated the PR to provide guidance about earlier versions as well.

BillWagner committed:
add location for earlier versions

mairaw commented on docs/csharp/language-reference/compiler-options/how-to-set-environment-variables-for-the-visual-studio-command-line.md:
```diff
@@ -40,20 +40,24 @@ translation.priority.ht:
+The VsDevCmd.bat file sets the appropriate environment variables to enable command-line builds. For more information about VsDevCmd.bat, see [Knowledge Base article Q248802](http://go.microsoft.com/fwlink/?LinkId=225042).  
+
+> [!NOTE]
+> The VsDevCmd.bat file is a new file delivered with Visual Studio 2017. Visual Studio 2015 and earlier versions used VSVARS32.bat for the same purpose. This file was stored in Program Files\Microsoft Visual Studio\*Version*\Common7\Tools or Program Files (x86)\Microsoft Visual Studio\*Version*\Common7\Tools.
```
this is rendering slightly funny. probably not your intention:
![image](https://user-images.githubusercontent.com/12971179/31039683-83875d20-a534-11e7-99c3-c5972e220839.png)

rpetrusha replied:
Each occurrence of "Program Files" should have a backslash in front of it (it's a subdirectory off the root).

mairaw commented on docs/csharp/language-reference/compiler-options/how-to-set-environment-variables-for-the-visual-studio-command-line.md:
```diff
@@ -40,20 +40,24 @@ translation.priority.ht:
+> The VsDevCmd.bat file is a new file delivered with Visual Studio 2017. Visual Studio 2015 and earlier versions used VSVARS32.bat for the same purpose. This file was stored in Program Files\Microsoft Visual Studio\*Version*\Common7\Tools or Program Files (x86)\Microsoft Visual Studio\*Version*\Common7\Tools.
   
- If the current version of Visual Studio is installed on a computer that also has an earlier version of Visual Studio, you should not run vsvars32.bat or vcvars32.bat from different versions in the same Command Prompt window.  
+If the current version of Visual Studio is installed on a computer that also has an earlier version of Visual Studio, you should not run VsDevCmd.bat and VSVARS32.BAT from different versions in the same Command Prompt window.
```
should you also add vcvars32 to this list as well? I'd also recommend complementing this with 
"Instead, you should run each command on its own window" or something.

mairaw commented on docs/csharp/language-reference/compiler-options/how-to-set-environment-variables-for-the-visual-studio-command-line.md:
```diff
@@ -40,20 +40,24 @@ translation.priority.ht:
+### To run VsDevCmd.BAT  
   
-1.  From the **Start** menu, open the **Developer Command Prompt for VS2012**.  
+1.  From the **Start** menu, open the **Developer Command Prompt for VS2017**.  It's in the **Visual Studio 2017** folder
```
missing period

mairaw commented on docs/csharp/language-reference/compiler-options/how-to-set-environment-variables-for-the-visual-studio-command-line.md:
```diff
@@ -40,20 +40,24 @@ translation.priority.ht:
+1.  From the **Start** menu, open the **Developer Command Prompt for VS2017**.  It's in the **Visual Studio 2017** folder
   
-2.  Change to the Program Files\Microsoft Visual Studio *Version*\Common7\Tools or Program Files (x86)\Microsoft Visual Studio *Version*\Common7\Tools subdirectory of your installation.  
+2.  Change to the Program Files\Microsoft Visual Studio\*Version*\*Offering*\Common7\Tools or Program Files (x86)\Microsoft Visual Studio\*Version*\*Offering*\Common7\Tools subdirectory of your installation.  (*Version* is *2017* for the current version. *Offering* is one of *Enterprise*, *Professional* or *Community*.)
```
weird styling here too
![image](https://user-images.githubusercontent.com/12971179/31039737-fc9c8ca8-a534-11e7-8990-ca205af87b2a.png)

@rpetrusha I think we usually call out versions in path in a different way no? like putting in tags or something.

rpetrusha replied:
@BillWagner, @mairaw, I don't think we've established a way to do it here. Previously, we used two conventions. One was <replaceable> tag/character style, which italicized the string using the current font. We also used a notation like `\Program Files\Microsoft Visual Studio\<version>\...`. 

Here, it would be clear if this appears in italics, as you'd intended it to, @BillWagner . The question is how to do that. The initial issues in each case are caused by the initial `\*` which is rendering as `*`, so I think I'd try either replacing * and * with <i> and </i>, or escaping the backslash that precedes ("\\*") and follows the asterisk ("*\\") to see what happens.

mairaw reviewed (changes requested):
LGTM. Left a few comments.

BillWagner committed:
update per feedback

I've fixed everything but the strange rendering for "version" and "offering".  @rpetrusha, what do you recommend here?

BillWagner committed:
fix format of italics in directories.

BillWagner commented:
@mairaw @rpetrusha I've addressed all the feedback. Can you give another look?

BillWagner committed:
missed one unescaped asterisk.

rpetrusha commented on docs/csharp/language-reference/compiler-options/how-to-set-environment-variables-for-the-visual-studio-command-line.md:
```diff
@@ -40,20 +40,24 @@ translation.priority.ht:
+The VsDevCmd.bat file sets the appropriate environment variables to enable command-line builds. For more information about VsDevCmd.bat, see [Knowledge Base article Q248802](http://go.microsoft.com/fwlink/?LinkId=225042).  
+
+> [!NOTE]
+> The VsDevCmd.bat file is a new file delivered with Visual Studio 2017. Visual Studio 2015 and earlier versions used VSVARS32.bat for the same purpose. This file was stored in \Program Files\Microsoft Visual Studio\\*Version*\Common7\Tools or Program Files (x86)\Microsoft Visual Studio\**Version*\Common7\Tools.
```
In the second version, you've added an asterisk instead of a backlash. (`\**Version` --> `\\*Version`)

BillWagner replied:
Just pushed a fix. I'll check it on the review site once it builds.

rpetrusha reviewed (commented):
@BillWagner, this looks good except for one omission.

BillWagner committed:
Fix a * and \ mistake

BillWagner commented:
@rpetrusha @mairaw I jus checked the review site, and I think it's ready.

rpetrusha commented:
@BillWagner, this looks good now. It's ready to merge. @mairaw, can you approve?

mairaw commented on docs/csharp/language-reference/compiler-options/how-to-set-environment-variables-for-the-visual-studio-command-line.md:
```diff
@@ -40,20 +40,24 @@ translation.priority.ht:
+### To run VsDevCmd.BAT  
   
-1.  From the **Start** menu, open the **Developer Command Prompt for VS2012**.  
+1.  From the **Start** menu, open the **Developer Command Prompt for VS2017**.  It's in the **Visual Studio 2017** folder.
```
nit: The name of the prompt contains a space VS 2017

mairaw reviewed (approved):
LGTM. A minor typo to fix left.

BillWagner committed:
add missing space
